rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -414,10 +458,21 @@ impl PendingMessage {
         pm
     }
 
-    fn on_reprepare(&mut self) -> PendingOperationResult {
+    fn on_reprepare<E: Debug>(
+        &mut self,
+        err: E,
+        reason: ReprepareReason,
+    ) -> PendingOperationResult {
         self.inc_attempts();
         self.submitted = false;
-        PendingOperationResult::Reprepare
+        warn!(error = ?err, "{}", reason.clone());
```

Comment:
if `err` is `""` does this mean that logs will have `error: ""`? Wonder if there's any clearer way to indicate no error actually happened - I often search "error" in logs and think it'd be nice to only show that when an error for sure happened